src/plugin/sdl/sdl.c

```diff
@@ -884,6 +884,14 @@ static void SDL_put_image(int x, int y, unsigned width, unsigned height)
 
   SDL_UpdateTexture(texture_buf, &rect, surface->pixels + offs,
       surface->pitch);
+
+  /* We have to synchronize resource updates across contexts manually.
+   *
+   * Could use a glSync object here, if available, but I doubt it is worth
+   * it. Uploading the screen data is the biggest chunk of work we do. */
+  if (update_ctx)
+      glFinish();
```

Comment:
Just what header and dependencies
does this require?
Looks like a can of worms...

Reply:
Good question. I assumed libSDL links to it, but I guess I shouldn't rely on it.

There's https://wiki.libsdl.org/SDL_GL_GetProcAddress . With it I can load it at runtime iff hwrend is enabled and we get a HW accelerated renderer.